### ISSUE TITLE
`Cannot use 'in' operator to search for '$ref' in undefined` when using 3 level nested object containing arrays

### ISSUE THREAD

**Issue by msb00003** [rjsf-team/react-jsonschema-form#3194]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

5.x (and whatever is presently deployed to the playground)

### Current Behavior

When using a schema which contains an object of arrays, which contains an object of arrays, which contains an object of arrays, on the playground (and `5.0.0-beta.10` locally) I am unable to add the third level down using the UI (only 2 levels of "+") and if I add formData in which has that third level I get `Cannot use 'in' operator to search for '$ref' in undefined`

Installed versions which reproduces this issue locally:
```
@rjsf/core@5.0.0-beta.10
@rjsf/utils@5.0.0-beta.10
@rjsf/validator-ajv6@5.0.0-beta.10
```

I'm unsure if this a bug or a missing feature I would like. I have not found it referenced in the documentation.

### Expected Behavior

That I would be able to add that third level set through the UI and consume a pre-existing dataset through the formData prop as I need it for part of a larger schema.

### Steps To Reproduce

Opening this to my understanding should have a third "+" under `thirdLevelArrayOfObjects` in order to add in that section: 
https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJ0b3BMZXZlbE9iamVjdCI6eyJmaXJzdEFycmF5T2ZPYmplY3RzIjpbeyJzZWNvbmRMZXZlbEFycmF5T2ZPYmplY3RzIjpbeyJzb21lT2JqZWN0SW5BcnJheSI6eyJ0aGlyZExldmVsQXJyYXlPZk9iamVjdHMiOlt7ImJvdHRvbVRoaW5nIjp7fX0seyJib3R0b21UaGluZyI6e319XX19XSwibmFtZSI6Ikdyb3VwMSJ9XX19LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsidG9wTGV2ZWxPYmplY3QiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsiZmlyc3RBcnJheU9mT2JqZWN0cyI6eyJ0eXBlIjoiYXJyYXkiLCJpdGVtcyI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJuYW1lIjp7InR5cGUiOiJzdHJpbmcifSwic2Vjb25kTGV2ZWxBcnJheU9mT2JqZWN0cyI6eyJ0eXBlIjoiYXJyYXkiLCJpdGVtcyI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJ0aGlzTGV2ZWxJc0ZpbmUiOnsidHlwZSI6InN0cmluZyJ9LCJzb21lT2JqZWN0SW5BcnJheSI6eyJ0eXBlIjoib2JqZWN0Iiwib25lT2YiOlt7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InRoaXJkTGV2ZWxBcnJheU9mT2JqZWN0cyI6eyJ0eXBlIjoiYXJyYXkiLCJpdGVtcyI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJib3R0b21UaGluZyI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJib3R0b21Qcm9wIjp7InR5cGUiOiJzdHJpbmcifX19fX19fX1dfX19fX19fX19fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZSI6ZmFsc2UsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZX19

For the `Cannot use 'in' operator to search for '$ref' in undefined` issue, open:
https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJ0b3BMZXZlbE9iamVjdCI6eyJmaXJzdEFycmF5T2ZPYmplY3RzIjpbeyJuYW1lIjoiR3JvdXAxIiwic2Vjb25kTGV2ZWxBcnJheU9mT2JqZWN0cyI6W3sidGhpc0xldmVsSXNGaW5lIjoiSWYgeW91IHJlbW92ZSBzb21lT2JqZWN0SW5BcnJheSB0aGlzIGRvZXMgbG9hZCIsInNvbWVPYmplY3RJbkFycmF5Ijp7InRoaXJkTGV2ZWxBcnJheU9mT2JqZWN0cyI6W3siYm90dG9tVGhpbmciOnsiYm90dG9tUHJvcCI6InNvbWVWYWx1ZSJ9fSx7ImJvdHRvbVRoaW5nIjp7ImJvdHRvbVByb3AiOiJzb21lVmFsdWUifX1dfX1dfV19fSwic2NoZW1hIjp7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InRvcExldmVsT2JqZWN0Ijp7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7ImZpcnN0QXJyYXlPZk9iamVjdHMiOnsidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsibmFtZSI6eyJ0eXBlIjoic3RyaW5nIn0sInNlY29uZExldmVsQXJyYXlPZk9iamVjdHMiOnsidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsidGhpc0xldmVsSXNGaW5lIjp7InR5cGUiOiJzdHJpbmcifSwic29tZU9iamVjdEluQXJyYXkiOnsidHlwZSI6Im9iamVjdCIsIm9uZU9mIjpbeyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJ0aGlyZExldmVsQXJyYXlPZk9iamVjdHMiOnsidHlwZSI6ImFycmF5IiwiaXRlbXMiOlt7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7ImJvdHRvbVRoaW5nIjp7InR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7ImJvdHRvbVByb3AiOnsidHlwZSI6InN0cmluZyJ9fX19fV19fX1dfX19fX19fX19fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZSI6ZmFsc2UsInJlYWRvbmx5IjpmYWxzZSwib21pdEV4dHJhRGF0YSI6ZmFsc2UsImxpdmVPbWl0IjpmYWxzZX19

### Environment

```markdown
- OS: MacOS 12.5.1
- Node: v16.14.0
- npm: 8.3.1

And reproducible on your playground as of 2022-10-11
```

**Comment by msb00003** [comment]
Issue seems to have been resolved in 5.0.0.beta.11
![Screenshot 2022-10-13 at 15 05 03](https://user-images.githubusercontent.com/12694845/195619260-62f1892f-1ccb-42f3-bb46-241fece46928.png)

**Comment by msb00003** [comment]
Cheers to whomever did that.
